spec/models/knight_spec.rb

```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Knight, type: :model do
```

Comment:
You should be testing the creations of all your children. This test would have yielded a failure due to lack of knight in type validation